scripts/validate.mjs

```diff
@@ -2,40 +2,76 @@ import path from 'path'
 import Ajv from 'ajv'
 import { readdirSync, readFileSync } from 'fs'
 
-const ajv = new Ajv()
-
 const SAMPLES_DIRECTORY = './samples'
 const SCHEMAS_DIRECTORY = './schemas'
 
-forEachFile(SCHEMAS_DIRECTORY, addSchema)
-forEachFile(SAMPLES_DIRECTORY, validateSample)
+validateSchemasObjectsPropertiesCase()
+validateSchemasIds()
+validateSamples()
 
-function addSchema(schemaPath) {
-  const schema = readJson(schemaPath)
+if (process.exitCode !== 0 && process.exitCode !== undefined) {
+  console.log('❌ Some validation errors were found')
+}
 
-  // We need to be careful about schema ids because they need to:
-  // * be unique, or else Ajv will throw
-  // * represent a path, as Ajv will use it to resolve $refs
-  // Here, we make sure that both requirements are respected.
-  const schemaId = computeSchemaIdFromSchemaPath(schemaPath)
-  if (schema.$id !== schemaId) {
-    console.log(`⚠️ Schema ${schemaPath} $id should be ${schemaId}`)
-    process.exitCode = 1
-  }
+function validateSchemasObjectsPropertiesCase() {
+  forEachFile(SCHEMAS_DIRECTORY, (schemaPath) => {
+    const schema = readJson(schemaPath)
 
-  ajv.addSchema(schema)
+    // RUM and telemetry schemas object properties should be snake_case, other schemas objects should
+    // be camelCase
```

Comment:
Question: would it be possible to have consistent casing across all schemas?

Reply:
It would be a very significant task. Changing snake_case to camelCase in RUM/Telemetry events would break pretty much everything including customer queries. Changing camelCase to snake_case in session replay is more easily achievable as it is not customer facing and to my knowledge only used in the frontend, but it would require @DataDog/rum-session-replay to handle both case styles (earlier versions of SDKs would still use camelCase) pretty much everywhere across the player and associated toolings.